playerbot/PlayerbotAI.cpp

```diff
@@ -5310,6 +5591,27 @@ void PlayerbotAI::InventoryIterateItems(IterateItemsVisitor* visitor, IterateIte
         InventoryIterateItemsInBuyBack(visitor);
 }
 
+std::vector<Item*> PlayerbotAI::GetInventoryAndEquipItems()

```

Comment:
fixed

Reply:
actually, that is not an improvement at all

Reply:
it returns a reference to an existing vector. what is the issue?

Reply:
there are no cases when one would already have a vector that would need to be populated, more convenient when the method returns a vector

Reply:
the problem with returning a vector is that it requires copying a vector. it's created once inside of the function and when the function returns it's created again and all the data inside of it is deep copied into the second new vector.
if you instead create the empty vector on the other side and just fill it within the function no additional copies are needed

Reply:
![image](https://github.com/user-attachments/assets/d5bb7a58-fda6-44c3-9467-9b11fbe36010)

Reply:
https://stackoverflow.com/questions/15704565/efficient-way-to-return-a-stdvector-in-c

Reply:
so it's preferred to return vector it seems

Reply:
check the codebase for precedence

Reply:
that procedure is inconvenient af, not for my use-case

Reply:
[insunaa](https://github.com/insunaa) I have created the method to not to use that one that requires visitor, as it is inconvenient af, this method should stay, and it would return vector